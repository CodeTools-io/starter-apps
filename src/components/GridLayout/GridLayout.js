import React, { useContext } from 'react';
import { Box, Grid, ResponsiveContext } from 'grommet';

export default function GridLayout({ children, gridProps = {}, ...props }) {
  const size = useContext(ResponsiveContext);

  return (
    <Box align="start" alignContent="start" justify="start" {...props}>
      <Grid
        columns={size !== 'small' ? 'small' : '100%'}
        rows="auto"
        align="start"
        alignContent="start"
        justify="start"
        justifyContent="start"
        gap={size}
        {...gridProps}
      >
        {children}
      </Grid>
    </Box>
  );
}
