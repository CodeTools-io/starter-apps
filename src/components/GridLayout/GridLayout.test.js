import React from 'react'
import GridLayout from './GridLayout'

describe('GridLayout', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <GridLayout>
        <p>Content for the GridLayout</p>
      </GridLayout>
    )

    expect(container).toMatchSnapshot()
  })
})

