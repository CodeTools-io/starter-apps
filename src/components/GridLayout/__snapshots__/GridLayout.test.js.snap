// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridLayout can render 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <div
      class="StyledGrid-sc-1wofa1l-0 hxvnVW"
    >
      <p>
        Content for the GridLayout
      </p>
    </div>
  </div>
</div>
`;
